libraries/helper.rb

```diff
@@ -4,22 +4,24 @@
 class Sumologic
   class ApiError < RuntimeError; end
 
-  def self.collector_exists?(node_name, email, pass)
+  def self.collector_exists?(node_name, email, pass, api_timeout = nil)
     collector = Sumologic::Collector.new(
       name: node_name,
       api_username: email,
-      api_password: pass
+      api_password: pass,
+      api_timeout: api_timeout
     )
     collector.exist?
   end
 
   class Collector
-    attr_reader :name, :api_username, :api_password
+    attr_reader :name, :api_username, :api_password, :api_timeout
 
     def initialize(opts = {})
       @name = opts[:name]
       @api_username = opts[:api_username]
       @api_password = opts[:api_password]
+      @api_timeout = opts.fetch(:api_timeout, nil)
```

Comment:
Fair point
Looks like the [default timeout](http://ruby-doc.org/stdlib-2.2.0/libdoc/net/http/rdoc/Net/HTTP.html#attribute-i-read_timeout) for Net::HTTP is 60 seconds so i will use that